postgres/lib/func.js

```diff
@@ -10,17 +11,64 @@ const StandardFunctions = {
   contains: (...args) => `(coalesce(strpos(${args}),0) > 0)`,
   indexof: (x, y) => `strpos(${x},${y}) - 1`, // sqlite instr is 1 indexed
   startswith: (x, y) => `strpos(${x},${y}) = 1`, // sqlite instr is 1 indexed
-  endswith: (x, y) => `substr(${x},length(${x}) + 1 - length(${y})) = ${y}`,
+  endswith: (x, y) => `substr(${x},length(${x}) + 1 - length(${y})) = ${y}`,  
 
   // Date and Time Functions
-  year: x => `date_part('year',(${x})::TIMESTAMP)`,
-  month: x => `date_part('month',(${x})::TIMESTAMP)`,
-  day: x => `date_part('day',(${x})::TIMESTAMP)`,
-  hour: x => `date_part('hour',(${x})::TIMESTAMP)`,
-  minute: x => `date_part('minute',(${x})::TIMESTAMP)`,
-  second: x => `date_part('second',(${x})::TIMESTAMP)`,
+  year: x => {
+    if (isTime(x.val)) throw new cds.error ({
```

Comment:
To much duplicate code. Also DB service does not create errors with HTTP error codes.